### DISCUSSION TITLE
Running CUDA Quantum on Colab/Kaggle

### DISCUSSION THREAD

**Discussion by essaathar** [NVIDIA/cuda-quantum#1183] [Q&A] [answered]
Was planning on implementing a Quantum Algorithm on CUDA. Since I do not have a GPU on my local machine, I've been using **Google Colab** and/or **Kaggle** for doing CUDA C Programming so far. Can I also similarly use **CUDA Quantum** in Colab/Kaggle, since I cannot find anything on this. Please let me know.

**Comment by bmhowe23** [comment, marked as answer]
https://github.com/NVIDIA/cuda-quantum/issues/1096 describes how to get the Python wheels working in Google Colab. Does that work for you? Or are you strictly asking about how to use the `nvq++` compiler in that environment?

**Reply by essaathar** [reply]
Thank you! That works for now. Now just two questions :
- How to use the `nvq++` compiler in Colab, if I choose to do CUDA QC in C++?
- Is there a tradeoff or an edge between doing CUDA QC in Python vs C++?

**Reply by bmhowe23** [reply]
I am not a Colab expert, but I am not aware of any way to run typical command-line activities in Colab, and that may limit your ability to use `nvq++` on Colab. That being said, you might want to consider https://nvidia.github.io/cuda-quantum/latest/install.html#dgx-cloud if you access to the command-line interface with cloud-provided GPUs.

Regarding Python vs C++, there is some documentation [here](https://nvidia.github.io/cuda-quantum/latest/using/examples.html) that may be useful for you. The examples show how to do similar items in both C++ and Python. The syntax is obviously different between the two languages, but both C++ and Python implementations have access to the same CUDA Quantum simulators and real QPU backends.

**Reply by essaathar** [reply]
Is it then not possible to use the C++ API on colab?

**Reply by bmhowe23** [reply]
That's correct ... I don't think it is possible to use the C++ API on Colab. However, you can use both the C++ API and Python API on [Nvidia Quantum Cloud](https://nvidia.github.io/cuda-quantum/latest/using/backends/nvqc.html#nvidia-quantum-cloud). NVQC is currently in Early Access phase; instructions on how to apply are provided on the linked documentation.
